include/plugin_interface/SC_World.h

```diff
@@ -103,6 +108,23 @@ struct World {
     nova::padded_rw_spinlock* mAudioBusLocks;
     nova::spin_lock* mControlBusLock;
 #endif
+
+#ifdef SC_BELA
+    BelaContext* mBelaContext;
+    BelaScope* mBelaScope;
+    uint32 mBelaMaxScopeChannels;
+    uint32 mBelaAnalogInputChannels;
+    uint32 mBelaAnalogOutputChannels;
+    uint32 mBelaDigitalChannels;
+    float mBelaHeadphoneLevel;
+    float mBelaPGAGainLeft;
+    float mBelaPGAGainRight;
+    bool mBelaSpeakerMuted;
+    float mBelaDACLevel;
+    float mBelaADCLevel;
+    uint32 mBelaNumMuxChannels;
+    uint32 mBelaPRU;
```

Comment:
I don't see a problem with that . The only fields used by ugens are `mBelaContext` and `mBelaScope`. Everything else is only used for startup options (command line options). There is also the possibility to add something like `void* Sc_getAudioContext()`  and `void* Sc_getCustomScope()` and go back to a vanilla `SC_World` also with Bela.

Reply:
This is rather tangential, but what I would find really nice is a way to add custom extensions to scsynth and make them available to plugins without breaking the existing plugin interface.

For example, the plugin interface could have an API method `(void *) (*fGetExtension)(const char *)` which allows to obtain a struct or function pointers by name, e.g. "Bela_Context", "Bela_Scope", "Bela_SomeCoolFunction". If the result a NULL, it means the extension is not available. Otherwise, the caller just has to cast the result to the correct type and store it somewhere for later use. I'm sure @scztt also has some thoughts about this :-)

Reply:
Yes!